dashboard.py

```diff
@@ -17,62 +16,141 @@
 panel.extension("vega")
 
 
-def get_test_source():
-    """
-    Crawl the tests directory and try to grab code for each test on a best-effort
-    basis. This relies on the tests being importable from this script, so the
-    environment should be similar enough that that is possible.
-    """
-    source: dict[str, str] = {}
-    files = glob.glob("tests/**/test_*.py", recursive=True)
-    for f in files:
-        try:
-            # Fragile!
-            mod = importlib.import_module(f.replace("/", ".")[: -len(".py")])
-            tests = [a for a in dir(mod) if a.startswith("test_")]
-            for test in tests:
-                if fn := getattr(mod, test, None):
-                    if not callable(fn):
-                        continue
-                    source[f[len("tests/") :] + "::" + test] = inspect.getsource(fn)
-        except BaseException:  # Some pytest exceptions inherit directly from BaseException
-            pass
-    return source
-
-
-source = get_test_source()
-
-
-def make_timeseries(originalname, df, spec) -> altair.Chart | None:
+class ChartSpec(NamedTuple):
+    field_name: str
+    field_desc: str
+    unit: str
+    baseline: str | None
+
+
+source: dict[str, str] = {}
+
+
+def load_test_source() -> None:
+    """Crawl the tests directory and try to grab code for each test. This relies on the
+    tests being importable from this script.
     """
-    Make a single timeseries altair chart for a given test.
+    for fname in glob.iglob("tests/**/test_*.py", recursive=True):
+        mod = importlib.import_module(fname.replace("/", ".")[: -len(".py")])
+        tests = [a for a in dir(mod) if a.startswith("test_")]
+        for test in tests:
+            if (func := getattr(mod, test, None)) and callable(func):
+                # FIXME missing decorators, namely @pytest.mark.parametrize
+                source[fname[len("tests/") :] + "::" + test] = inspect.getsource(func)
```

Comment:
I'd rather continue to be defensive here with a `try`/`except`

Reply:
Why? It makes a lot more sense to me to have the whole test suite crash if someone adds a new import in a test file and doesn't add it to `ci/environment-dashboard.yml`.

Reply:
Because I view this code-scraping to be a fairly fragile hack, and weird stuff can happen on import, especially when pytest is involved. For example, a `pytest.skip` at the module level raises an exception, and I don't want to force everyone to do non-idiomatic pytest stuff on account of this static report. To me, having a missing code snippet for a test is not a big deal.

Reply:
I'm also unhappy about having to add stuff to `ci/environment-dashboard.yml` in the first place to just be able to use `inspect`. If you have ideas about how to get snippets without relying on importing, I'm certainly open to it. But this seemed preferable to trying to parse `.py` files to me.

Reply:
Partially reverted